### PR TITLE
Enable TypeScript strict mode

### PR DESCRIPTION
**Pull request by gustavohenke**
As title says. Addressing https://github.com/open-cli-tools/concurrently/pull/372#issuecomment-1277604405 😉

### CODE REVIEW

File: bin/concurrently.spec.ts [hunk 1]
```diff
@@ -82,8 +80,8 @@ const run = (args: string, ctrlcWrapper?: boolean) => {
     });
 
     const exit = Rx.firstValueFrom(
-        Rx.fromEvent(child, 'exit').pipe(
-            map((exit: [number | null, NodeJS.Signals | null]) => {
+        Rx.fromEvent<[number | null, NodeJS.Signals | null]>(child, 'exit').pipe(
```
Comment (gustavohenke) [inline bin/concurrently.spec.ts:83]: Sorry, I realise this is more trouble for #343.
I think we could get away with a `resultSelector` arg that casts the value though

File: src/command.ts [hunk 2]
```diff
@@ -92,7 +92,7 @@ export class Command implements CommandInfo {
     readonly command: string;
 
     /** @inheritdoc */
-    readonly prefixColor: string;
+    readonly prefixColor?: string;
```
Comment (gustavohenke) [inline src/command.ts:95]: Actually matches the `CommandInfo` interface.

File: src/command.ts [hunk 3]
```diff
@@ -112,8 +112,9 @@ export class Command implements CommandInfo {
     killed = false;
     exited = false;
 
+    /** @deprecated */
```
Comment (gustavohenke) [inline src/command.ts:115]: Let's remove this in a future PR 🙂

File: src/completion-listener.ts [hunk 4]
```diff
@@ -92,11 +92,15 @@ export class CompletionListener {
...
+                        ? this.emitWithScheduler(Rx.of(exitInfos))
+                        : this.emitWithScheduler(Rx.throwError(exitInfos))
                 ),
                 take(1)
             )
         );
     }
+
+    private emitWithScheduler<O>(input: Rx.Observable<O>): Rx.Observable<O> {
+        return this.scheduler ? input.pipe(Rx.observeOn(this.scheduler)) : input;
```
Comment (gustavohenke) [inline src/completion-listener.ts:104]: So RxJS types are a bit finicky. When you don't have a scheduler, you can't pass the arg at all to functions that accept it.

File: src/concurrently.ts [hunk 5]
```diff
@@ -30,7 +30,7 @@ const defaults: ConcurrentlyOptions = {
  * If value is a string, then that's the command's command line.
  * Fine grained options can be defined by using the object format.
  */
-export type ConcurrentlyCommandInput = string | Partial<CommandInfo>;
+export type ConcurrentlyCommandInput = string | ({ command: string } & Partial<CommandInfo>);
```
Comment (gustavohenke) [inline src/concurrently.ts:33]: bit weird to allow a command not to have a command...

**Comment by gustavohenke** [comment]
I forgot to rebase before pushing.
One moment please

**Comment by gustavohenke** [comment]
Let me know of any changes you think might be useful, @paescuj 🙂

File: bin/concurrently.spec.ts (hunk 1 above)
Comment (paescuj) [reply bin/concurrently.spec.ts:83]: No problem! We can address this in #343. We probably have to rely on `map` though as `resultSelector` is deprecated: https://rxjs.dev/deprecations/resultSelector

File: src/command.ts (hunk 3 above)
Comment (paescuj) [reply src/command.ts:115]: Good idea 👍

File: src/completion-listener.ts (hunk 4 above)
Comment (paescuj) [reply src/completion-listener.ts:104]: I see 👍 Good solution!

File: src/concurrently.ts (hunk 5 above)
Comment (paescuj) [reply src/concurrently.ts:34]: Nice 😄

**Comment by paescuj** [comment]
Thanks for taking this on and sorry for my late reply! Looks solid 👍😃

**Pull request merged**